File: Templates/Frameworks-Samples/Csla/CSharp/PetShop/Source/PetShop.UI/Controls/WishListControl.ascx.cs
using System;
using System.Web.UI.WebControls;

using PetShop.Business;

namespace PetShop.UI.Controls
{
    public partial class WishListControl : System.Web.UI.UserControl
    {
        /// <summary>
        /// Handle Page load event
        /// </summary>
        protected void Page_PreRender(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(Page.User.Identity.Name))
                Response.Redirect("~/SignIn.aspx");

            if (!IsPostBack)
            {
                BindCart();
            }
        }

        /// <summary>
        /// Bind repeater to Cart object in Profile
        /// </summary>
        private void BindCart()
        {
            Profile profile = Profile.GetProfile(Page.User.Identity.Name);
            if (!string.IsNullOrEmpty(profile.Username))
            {
                Business.CartList wishList = profile.WishList;
                if (wishList.Count > 0)
                {
                    repWishList.DataSource = wishList;
                    repWishList.DataBind();
                }
                else
                {
                    repWishList.Visible = false;
                    lblMsg.Text = "Your wish list is empty.";
                }
            }
        }

        /// <summary>
        /// Handler for Delete/Move buttons
        /// </summary>
        protected void CartItem_Command(object sender, CommandEventArgs e)
        {
            Profile profile = Profile.GetProfile(Page.User.Identity.Name);
            if (!string.IsNullOrEmpty(profile.Username))
            {
                switch (e.CommandName)
                {
                    case "Del":
                        profile.WishList.Remove(e.CommandArgument.ToString());
                        break;
                    case "Move":
                        profile.WishList.Remove(e.CommandArgument.ToString());
                        profile.ShoppingCart.Add(e.CommandArgument.ToString(), profile.UniqueID, true);
                        break;
                }

                profile = profile.Save();
            }

            BindCart();
        }
    }
}

File: tags/CSLA-3.0.0-Samples/VisualBasic/PetShop/Source/PetShop.UI/Controls/WishListControl.ascx.vb
Imports PetShop.Business

Partial Public Class WishListControl
    Inherits System.Web.UI.UserControl

    ''' <summary>
    ''' Handle Page load event
    ''' </summary>
    Protected Sub Page_PreRender(ByVal sender As Object, ByVal e As EventArgs)
        If Not IsPostBack Then
            BindCart()
        End If
    End Sub

    ''' <summary>
    ''' Bind repeater to Cart object in Profile
    ''' </summary>
    Private Sub BindCart()
        Dim profile As Profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name)

        Dim wishList As Business.CartList = profile.WishList
        If wishList.Count > 0 Then
            repWishList.DataSource = wishList
            repWishList.DataBind()
        Else
            repWishList.Visible = False
            lblMsg.Text = "Your wish list is empty."
        End If
    End Sub

    ''' <summary>
    ''' Handler for Delete/Move buttons
    ''' </summary>
    Protected Sub CartItem_Command(ByVal sender As Object, ByVal e As CommandEventArgs)
        Dim profile As Profile = ProfileManager.Instance.GetCurrentUser(Page.User.Identity.Name)

        Select Case e.CommandName
            Case "Del"
                profile.WishList.Remove(e.CommandArgument.ToString())
                Exit Select
            Case "Move"
                profile.WishList.Remove(e.CommandArgument.ToString())
                profile.ShoppingCart.Add(e.CommandArgument.ToString(), profile.UniqueID, True)
                Exit Select
        End Select

        profile = profile.Save()

        BindCart()
    End Sub

End Class